test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1

```diff
@@ -16,4 +16,19 @@ Describe 'ConvertTo-Json' -tags "CI" {
         $jsonFormat | Should Match '"TestValue2": 78910'
         $jsonFormat | Should Match '"TestValue3": 99999'
     }
+
+	It "StopProcessing should succeed" {
+        $tmpFile = Join-Path $TestDrive "test.txt"
+        Set-Content -Path $tmpFile -Value "hello"
+        $ps = [PowerShell]::Create()
+        $null = $ps.AddCommand("Get-Content")
+        $null = $ps.AddParameter("Path", $tmpFile)
+        $null = $ps.AddCommand("ConvertTo-Json")
+        $null = $ps.AddParameter("Depth", 10)
+        $null = $ps.BeginInvoke()
+        Start-Sleep -Milliseconds 100
+        $null = $ps.Stop()
```

Comment:
I tested this with 6.0.1 and a less extreme example and it passed. In a regression this could potentially block the pester thread for a long time and give a false pass. Perhaps we should do something closer to this:

```powershell
$null = $ps.BeginStop($null, $null)
Start-Sleep -Milliseconds 100
$ps.InvocationStateInfo.State | Should -Be "Stopped"
$ps.Dispose()
```

Reply:
Will use your test above